### PR TITLE
Fix condition in repo/default for debian or ubuntu

### PR DESCRIPTION
**Pull request by maximenoel8**
## What does this PR change?
Currently, part of repo/default is not executed for ubuntu and debian clients.
Fix the condition to correctly be executed for ubuntu and debian clients.

### CODE REVIEW

**Review by nodeg — CHANGES_REQUESTED** [review]
The condition seems correct and is only valid for Ubuntu. So no changes required.

Ubuntu:
```bash
root@suma-bv-50-minssh-ubuntu2204:~# venv-salt-call --local grains.get os_family
local:
    Debian
root@suma-bv-50-minssh-ubuntu2204:~# venv-salt-call --local grains.get os
local:
    Ubuntu 
```

```bash
Debian:
root@suma-bv-50-min-debian11:~# venv-salt-call --local grains.get os_family
local:
    Debian
root@suma-bv-50-min-debian11:~# venv-salt-call --local grains.get os
local:
    Debian
```

File: salt/repos/default.sls [hunk 1]
```diff
@@ -694,7 +694,7 @@ clean_repo_metadata:
 
 {% endif %} {# grains['os_family'] == 'RedHat' #}
 
-{% if grains['os_family'] == 'Debian' and grains['os'] == 'Ubuntu' %}
+{% if grains['os_family'] == 'Debian' or grains['os'] == 'Ubuntu' %}
```
Comment (srbarrios) [inline salt/repos/default.sls:697]: If we really want to run some salt states for both.
We can just use the family name, without filtering by os name.

```suggestion
{% if grains['os_family'] == 'Debian' %}
```

But I think the tools_update_repo must diverge by os name.

Comment (nodeg) [reply salt/repos/default.sls:697]: Taking a look at the complete `if` statement, I think this is only for Ubuntu and not both. Everything only mentions Ubuntu. See
https://github.com/uyuni-project/sumaform/blob/fa37bc5f3d5862dda442170ba4c9d8bfa18e1eb2/salt/repos/default.sls#L697-L795

**Comment by nodeg** [comment]
With https://github.com/uyuni-project/sumaform/pull/1575 being merged before this PR, we can merge this one as is.

**Comment by nodeg** [comment]
I think we still need this change, @maximenoel8.

**Pull request closed without merging**